packages/popup/src/hooks/useApi.js

```diff
@@ -222,30 +214,130 @@ export const useAddressType = address => {
   return type
 }
 
-export const useDbHomeAssets = () => {
-  const {eid: accountId} = useCurrentAccount()
-  const {eid: networkId} = useCurrentNetwork()
-  const {data: homeAssets} = useRPC(
-    [WALLETDB_HOME_PAGE_ASSETS, accountId, networkId],
-    undefined,
+export const useDbRefetchBalance = param => {
+  useRPC([WALLETDB_REFETCH_BALANCE], param ? {...param} : undefined)
+  useRPC(
+    [WALLETDB_REFETCH_BALANCE, 'REFETCH_ALL'],
+    {type: 'all', allNetwork: true},
+    {refreshInterval: 180000},
+  )
+}
+
+export const useCurrentNetworkToken = ({fuzzy, addressId}) => {
+  useDbRefetchBalance({type: 'all'})
+  const {
+    data: {
+      network: {eid: networkId},
+    },
+  } = useCurrentAddress()
+
+  return useRPC(
+    networkId
+      ? [
+          QUERY_TOKEN,
+          'useCurrentNetworkToken',
+          networkId,
+          addressId,
+          fuzzy || '',
+        ]
+      : null,
+    {fuzzy, addressId, networkId},
+    {fallbackData: []},
+  )
+}
+export const useCurrentAddressTokens = () => {
+  const {
+    data: {eid: addressId},
+  } = useCurrentAddress()
+  useDbRefetchBalance()
+  return useRPC(
+    addressId ? [QUERY_ADDRESS, 'useCurrentAddressTokens', addressId] : null,
     {
-      fallbackData: {},
+      addressId,
+      g: {token: 1},
+    },
+    {
+      fallbackData: [],
+      postprocessSuccessData: d => (d?.token || []).map(t => t.eid),
     },
   )
-  useDbRefetchBalance({type: 'all'})
-  return homeAssets
 }
-
-export const useDbRefetchBalance = param => {
-  useRPC([WALLETDB_REFETCH_BALANCE], param ? {...param} : undefined)
+export const useSingleTokenInfoWithNativeTokenSupport = tokenId => {
+  if (tokenId === 'native' || tokenId === '0x0') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const {ticker} = useCurrentAddress().data.network
+    ticker.logoURI = ticker.iconUrls?.[0]
+    ticker.decimals = 18
+    return ticker
+  }
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  return useRPC(
+    tokenId ? [QUERY_TOKEN, 'useSingleTokenInfo', tokenId] : null,
+    {
+      tokenId,
+      g: {name: 1, address: 1, symbol: 1, decimals: 1, logoURI: 1},
+    },
+    {fallbackData: {}},
+  ).data
 }
 
-export const useDbAddTokenList = () => {
-  const {data: addTokenListData} = useRPC([WALLETDB_ADD_TOKEN_LIST, 'all'], {
-    type: 'all',
-  })
-  useDbRefetchBalance()
-  return addTokenListData
+export const useSingleAddressTokenBalanceWithNativeTokenSupport = ({
+  addressId,
+  tokenId,
+}) => {
+  if (!addressId) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    addressId = useCurrentAddress().data.eid
+  }
+  if (tokenId === '0x0' || tokenId === 'native') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useRPC(
+      addressId
+        ? [QUERY_ADDRESS, 'useAddressTokenBalanceSupportNativeToken', addressId]
```

Comment:
How about `useNativeTokenBalance` here?

Reply:
the key indicating it's called inside `useSingleAddressTokenBalanceWithNativeTokenSupport`.
there are too many assumptions here in fallback data and postprocessSuccessData
`useNativeTokenBalance` is too general